internal/tools/changes/repository.go

```diff
@@ -0,0 +1,57 @@
+package changes
+
+import (
+	"fmt"
+	"path/filepath"
+)
+
+// Repository is a representation of a git repository containing multiple Go modules.
+type Repository struct {
+	RootPath string    // RootPath is the path to the root of the repository.
+	Metadata *Metadata // Metadata is the repository's .changes metadata.
+}
+
+// NewRepository loads the repository at the given path.
+func NewRepository(path string) (*Repository, error) {
+	metadata, err := LoadMetadata(filepath.Join(path, ".changes"))
```

Comment:
You probably want to put the file names into constants instead of referring to them inline.